program-runtime/src/execution_budget.rs

```diff
@@ -0,0 +1,252 @@
+#[cfg(feature = "dev-context-only-utils")]
+use qualifier_attr::qualifiers;
+use {solana_program_entrypoint::HEAP_LENGTH, std::num::NonZeroU32};
+
+#[cfg(feature = "frozen-abi")]
+impl ::solana_frozen_abi::abi_example::AbiExample
+    for crate::execution_budget::SVMTransactionExecutionBudget
+{
+    fn example() -> Self {
+        // ComputeBudget is not Serialize so just rely on Default.
+        crate::execution_budget::SVMTransactionExecutionBudget::default()
+    }
+}
+
+/// Max instruction stack depth. This is the maximum nesting of instructions that can happen during
+/// a transaction.
+pub const MAX_INSTRUCTION_STACK_DEPTH: usize = 5;
+
+/// Max call depth. This is the maximum nesting of SBF to SBF call that can happen within a program.
+pub const MAX_CALL_DEPTH: usize = 64;
+
+/// The size of one SBF stack frame.
+pub const STACK_FRAME_SIZE: usize = 4096;
+
+pub const MAX_COMPUTE_UNIT_LIMIT: u32 = 1_400_000;
+
+/// Roughly 0.5us/page, where page is 32K; given roughly 15CU/us, the
+/// default heap page cost = 0.5 * 15 ~= 8CU/page
+pub const DEFAULT_HEAP_COST: u64 = 8;
+pub const DEFAULT_INSTRUCTION_COMPUTE_UNIT_LIMIT: u32 = 200_000;
+// SIMD-170 defines max CUs to be allocated for any builtin program instructions, that
+// have not been migrated to sBPF programs.
+pub const MAX_BUILTIN_ALLOCATION_COMPUTE_UNIT_LIMIT: u32 = 3_000;
+pub const MAX_HEAP_FRAME_BYTES: u32 = 256 * 1024;
+pub const MIN_HEAP_FRAME_BYTES: u32 = HEAP_LENGTH as u32;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct SVMTransactionExecutionBudget {
```

Comment:
this thing is 360 bytes, we should probably not be implementing `Copy` on it.

Reply:
realize you just moved it, but we should also fix this imo

Reply:
Yes, agree on removing copy. I tried removing it in my workspace, and changes are not trivial. So a separate PR will be better.